code/kobe.py

```diff
@@ -1107,32 +1117,543 @@ def kobe_shadows(df_input,cdpp):
     
     return df_kobe_output, [nsystem_nonzero_views1,nsystem_nonzero_views2,nplanet_transiting]
 
+def calculate_ntransits_phase(df_kobe_output):
+    
+    """
+    Defined with KOBE version 2.0
+    
+    This function calculates the number of transits seen in the survey duration t_obs. This
+    is done assuming a random phase for each system, to account for the impact of the planet's
+    position on the number of transits
+
+    NOTE
+    ----
+    1. In the same system, planets are given the same phase at the time of observation.
+       Function updates the number of transits in the function kobe_transits.
+    
+    Parameters
+    ----------
+    df_kobe_output : pandas dataframe
+        dataframe of planets that transit (potentially detectable)
+        columns: all from input, some are added
+
+    Output 
+    ------
+    number_transits : numpy array 
+        array of values that give the number of transits a planet will allow to be observed
+        in the survey duration
+
+    """
+    # for each system in the KOBE-Shadows Catalogue, create an array 'thetas_system'
+    # which assigns one single random value of phase for each of the planets in that particular system  
+    number_systems = df_kobe_output['kobe_system_id'].unique().shape[0]
+    temp_df = df_kobe_output.groupby(['kobe_system_id']).size().reset_index(name='mul')
+    multiplicity = temp_df['mul']
+    listofones = np.split(np.ones(multiplicity.sum()), multiplicity.cumsum())
+    
+    theta = 360*np.random.random(number_systems)
+    thetas = [x[0]*x[1] for x in zip(theta, listofones)]
+    thetas_system = np.concatenate(thetas)
+
+    # Calculate number of transits using an average angular speed, the survey duration 
+    # and the above calculated random value of phase for each of the planets.
+    # To account for the different positions planets might have in one system, use the true anomaly
+    ang_speed = 360/df_kobe_output[col_period]
+    ang_dist = ang_speed*t_obs
+    theta_phase = (thetas_system + df_kobe_output['kobe_observer_azimuth'])%360 + ang_dist
```

Comment:
Umm.. why are we using the modulo operator here? And I'm worried about the units. I'm sure the kobe_observer_azimuth is in radians while your random theta seems to be in degrees?